sources/core.ts

```diff
@@ -867,6 +968,19 @@ export class CommandBuilder<Context> {
 
       for (let t = 0; t < path.length; ++t) {
         const nextPathNode = injectNode(machine, makeNode());
+
+        // TODO: find a way to not leak command into the core
+        registerDynamic(machine, lastPathNode, `isCompletion`, NODE_SUCCESS, [`setCompletion`, CompletionType.PathSegment, (request, command) => {
```

Comment:
I feel like it would make more sense to make the state machine serializable again in a different PR when we actually start using the serializability property - the implementation would vary depending on the specific use case.

For example, I could have a workaround to not passing callbacks but it would only make the code more complicated for no reason for now, and if in the future we'd want to pre-generate the state machine for example and hydrate it at runtime the implementation would be incompatible and would have to be changed anyways.